### PR TITLE
Replace cluster scoped AppCatalog with namespace scoped Catalog CRD

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/roadmap/issues/219

This updates the app commands to use the new Catalog CRD. During the migration period we will maintain both appcatalog and catalog CRs for our public catalogs.

### CODE REVIEW

**Review by axbarsan — APPROVED** [review]
LGTM

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Comment by rossf7** [comment]
We want to remove the selector flag since this is not the usual kubectl approach and can be confusing to users.

Instead we can exclude by namespace as our internal catalogs will be in the giantswarm namespace.

```
-l, --selector string                Specify label selector(s) to filter Apps by. (default "application.giantswarm.io/catalog-visibility=public")
```

cc @marians

File: cmd/get/catalogs/flag.go [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-	LabelSelector string
+	AllNamespaces bool
 
 	config genericclioptions.RESTClientGetter
 	print  *genericclioptions.PrintFlags
 }
 
 func (f *flag) Init(cmd *cobra.Command) {
-	cmd.Flags().StringVarP(&f.LabelSelector, flagLabelSelector, "l", publicCatalogSelector, "Specify label selector(s) to filter Apps by.")
+	cmd.Flags().BoolVarP(&f.AllNamespaces, flagAllNamespaces, "A", false, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
```
Comment (rossf7) [inline cmd/get/catalogs/flag.go:20]: Label selector is replaced with all namespaces flag.

File: pkg/data/domain/catalog/service.go [hunk 2]
```diff
@@ -39,68 +39,72 @@ func New(config Config) (Interface, error) {
...
 				return nil, microerror.Mask(noResourcesError)
 			}
 		}
 
-		for _, appCatalog := range appCatalogs.Items {
-			a := AppCatalog{
-				CR: omitManagedFields(appCatalog.DeepCopy()),
+		for _, catalog := range catalogs.Items {
+			// We hide catalog CRs from the giantswarm namespace by
+			// default as these are internal.
```
Comment (rossf7) [inline pkg/data/domain/catalog/service.go:83]: If -A is true we hide internal catalogs in the giantswarm namespace.

You can still use -n giantswarm if you have access.

**Review by rossf7 — COMMENTED** [review]
@marians Changes to remove label selector are done.

We still need to have the Catalog CRD deployed to production MCs before this can be merged.

File: pkg/template/catalog/catalog.go [hunk 3]
```diff
@@ -17,48 +17,49 @@ type Config struct {
...
 			APIVersion: "application.giantswarm.io/v1alpha1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: config.Name,
-			Labels: map[string]string{
-				"app-operator.giantswarm.io/version":     "1.0.0",
-				"application.giantswarm.io/catalog-type": "awesome",
-			},
+			Name:      config.Name,
+			Namespace: config.Namespace,
```
Comment (rossf7) [inline pkg/template/catalog/catalog.go:48]: Dropping these labels as we don't need them.

**Review by axbarsan — APPROVED** [review]

**Pull request merged**